### ISSUE TITLE
When running program, returning permanentFail error.

### ISSUE THREAD

**Issue by saba-m-a** [jorvis/GALES#6]
Error still persisting despite updating to latest local version of GALES code and Docker images for biocode and GALES.

```
Final process status is permanentFail
ERROR: Return code 1 when running the following command: cwl-runner --outdir /tmp/demo /tmp/demo/prok-annotation-cheetah.cwl /tmp/demo/prok-annotation-cheetah.json

```

**Comment by jorvis** [comment]
Can you post the full stack somewhere like [Gist](https://gist.github.com/)?  That one line isn't enough to track down what happened.

**Comment by DavidStreid** [comment]
I also received the same error on the "rapsearch2" step. Below is my output & stacktrace - 
```
[step rapsearch2] start
Unexpected exception
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/cwltool/workflow.py", line 840, in job
    **kwargs):
  File "/Library/Python/2.7/site-packages/cwltool/command_line_tool.py", line 374, in job
    builder.pathmapper = self.makePathMapper(reffiles, builder.stagedir, **make_path_mapper_kwargs)
  File "/Library/Python/2.7/site-packages/cwltool/command_line_tool.py", line 233, in makePathMapper
    separateDirs=kwargs.get("separateDirs", True))
  File "/Library/Python/2.7/site-packages/cwltool/pathmapper.py", line 229, in __init__
    self.setup(dedup(referenced_files), basedir)
  File "/Library/Python/2.7/site-packages/cwltool/pathmapper.py", line 282, in setup
    self.visit(fob, stagedir, basedir, copy=fob.get("writable"), staged=True)
  File "/Library/Python/2.7/site-packages/cwltool/pathmapper.py", line 271, in visit
    self.visitlisting(obj.get("secondaryFiles", []), stagedir, basedir, copy=copy, staged=staged)
  File "/Library/Python/2.7/site-packages/schema_salad/sourceline.py", line 157, in __exit__
    raise self.makeError(six.text_type(exc_value))
ValidationException: ../../../../../../../../tmp/demo/prok-annotation-cheetah.json:38:33: [Errno 2] No such file or directory: '/tmp/demo//../../dbs/uniprot_sprot_all_and_trembl_characterized.20171025.rapsearch2.db'
Cannot make scatter job: ../../../../../../../../tmp/demo/prok-annotation-cheetah.json:38:33: [Errno 2] No such file or directory: '/tmp/demo//../../dbs/uniprot_sprot_all_and_trembl_characterized.20171025.rapsearch2.db'
[step rapsearch2] completed permanentFail
[workflow prok-annotation-cheetah.cwl] completed permanentFail
{
    "hmmscan_htab_files": null,
    "aragorn_raw_output": {
        "checksum": "sha1$2568a5087e8a8508e66ffc41fc47140f9378186d",
        "basename": "aragorn.out",
        "location": "file:///tmp/demo/aragorn.out",
        "path": "/tmp/demo/aragorn.out",
        "class": "File",
        "size": 31172
    },
    "prodigal_protein_file": {
        "checksum": "sha1$3bc1c7d4cb903cac26baf8c0cb895a8bdcc48052",
        "basename": "prodigal.annotation.faa",
        "location": "file:///tmp/demo/prodigal.annotation.faa",
        "path": "/tmp/demo/prodigal.annotation.faa",
        "class": "File",
        "size": 1363301
    },
    "prodigal_annot_file": {
        "checksum": "sha1$99b653209c88c7d265c6c0c1f82ad72e81225c28",
        "basename": "prodigal.annotation.gff",
        "location": "file:///tmp/demo/prodigal.annotation.gff",
        "path": "/tmp/demo/prodigal.annotation.gff",
        "class": "File",
        "size": 788311
    },
    "merged_gff3": {
        "checksum": "sha1$e663017b7d43e4235070f39fc1097811e52bc5fe",
        "basename": "merged_structures.gff3",
        "location": "file:///tmp/demo/merged_structures.gff3",
        "path": "/tmp/demo/merged_structures.gff3",
        "class": "File",
        "size": 4387030
    },
    "prodigal_protein_fasta": {
        "checksum": "sha1$d4e003d1698e5f7a60def9e9458c562fd6f16574",
        "basename": "prodigal2fasta.faa",
        "location": "file:///tmp/demo/prodigal2fasta.faa",
        "path": "/tmp/demo/prodigal2fasta.faa",
        "class": "File",
        "size": 1030344
    },
    "prodigal_gff3": {
        "checksum": "sha1$6ff7eff9b640e5d0a2bbb56a2f65cb754698cb03",
        "basename": "prodigal.annotation.gff3",
        "location": "file:///tmp/demo/prodigal.annotation.gff3",
        "path": "/tmp/demo/prodigal.annotation.gff3",
        "class": "File",
        "size": 1952418
    },
    "rapsearch2_m8_files": [
        null
    ],
    "hmmscan_raw_files": null,
    "barrnap_gff_output": {
        "checksum": "sha1$7ae5dc1cce1c3500e0cac62afbe926c08e8bb620",
        "basename": "barnapp.gff",
        "location": "file:///tmp/demo/barnapp.gff",
        "path": "/tmp/demo/barnapp.gff",
        "class": "File",
        "size": 16
    },
    "tmhmm_raw_files": null,
    "fasta_files": [
        {
            "checksum": "sha1$d4e003d1698e5f7a60def9e9458c562fd6f16574",
            "basename": "prodigal2fasta.faa.part1",
            "location": "file:///tmp/demo/prodigal2fasta.faa.part1",
            "path": "/tmp/demo/prodigal2fasta.faa.part1",
            "class": "File",
            "size": 1030344
        }
    ],
    "attributor_files": null,
    "attributor_output_config": null
}
Final process status is permanentFail
ERROR: Return code 1 when running the following command: cwl-runner --outdir /tmp/demo/ /tmp/demo//prok-annotation-cheetah.cwl /tmp/demo//prok-annotation-cheetah.json
```

**Comment by seann27** [comment]
I have also experienced this issue. I placed the full trace file in this gist:

https://gist.github.com/seann27/251c4336d26f231c84be01e1821b0424

**Comment by jorvis** [comment]
Can I see the command line you used to run it?  It appears that the -rd directory you passed doesn't actually have the reference files within it.  I clearly need to add this to the beginning of the script so that it checks before starting.  Here's what it attempted based on what you passed:

/tmp/demo//../../dbs/uniprot_sprot_all_and_trembl_characterized.20171025.rapsearch2.db

**Comment by DavidStreid** [comment]
I believe I have all of the needed reference files. Here is a gist showing my directory structure and complete log output - https://gist.github.com/DavidStreid/bac222802d541d535f3bbac593364260

**Comment by seann27** [comment]
When I used the full path of the dbs directory, it worked. ~/GALES/bin/dbs instead of typing dbs

**Comment by jorvis** [comment]
Glad it worked for you.  Just so you know, for other software which require full paths using the ~ symbol may not work.  A full path begins with the / symbol.

**Comment by JenHauen16** [comment]
Even with full paths, I can not get past the cwl-runner output directory not found.

jenha@DESKTOP-NJHO60U ~/GALES/GALES/bin
$ ls
Darwin  download_reference_data              GCA_002420305.1_ASM242030v1_genomic.fna  launch_interactive  list_gce_bucket.py  prok-annotation-cheetah.cwl   report_stats.py    view_annotation
dbs     filter_blast_ev_db_by_accessions.py  generate_pipeline_list.py                Linux               output              prok-annotation-cheetah.json  run_prok_pipeline

jenha@DESKTOP-NJHO60U ~/GALES/GALES/bin
$ /home/jenha/GALES/GALES/bin/run_prok_pipeline -i /home/jenha/GALES/GALES/bin/GCA_002420305.1_ASM242030v1_genomic.fna -od /home/jenha/GALES/GALES/bin/output/ -v cheetah -rd /home/jenha/GALES/GALES/bin/dbs/
Starting pipeline.  If this is your first time running this pipeline there will be some latency as the necessary Docker images are downloaded. This is a one-time cost.
Running command: cwl-runner --outdir /home/jenha/GALES/GALES/bin/output /home/jenha/GALES/GALES/bin/output/prok-annotation-cheetah.cwl /home/jenha/GALES/GALES/bin/output/prok-annotation-cheetah.json
C:\Users\jenha\AppData\Local\Programs\Python\Python36\Scripts\cwl-runner 1.0.20190228155703
Traceback (most recent call last):
  File "c:\users\jenha\appdata\local\programs\python\python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\jenha\appdata\local\programs\python\python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\jenha\AppData\Local\Programs\Python\Python36\Scripts\cwl-runner.exe\__main__.py", line 9, in <module>
  File "c:\users\jenha\appdata\local\programs\python\python36\lib\site-packages\cwltool\main.py", line 632, in main
    fetcher_constructor=loadingContext.fetcher_constructor)
  File "c:\users\jenha\appdata\local\programs\python\python36\lib\site-packages\cwltool\load_tool.py", line 89, in resolve_tool_uri
    raise ValidationException("Not found: '%s'" % argsworkflow)
schema_salad.validate.ValidationException: Not found: '/home/jenha/GALES/GALES/bin/output/prok-annotation-cheetah.cwl'
ERROR: Return code 1 when running the following command: cwl-runner --outdir /home/jenha/GALES/GALES/bin/output /home/jenha/GALES/GALES/bin/output/prok-annotation-cheetah.cwl /home/jenha/GALES/GALES/bin/output/prok-annotation-cheetah.json

**Comment by jorvis** [comment]
And did you say that when you list that directory the files are in fact there?

$ ls /home/jenha/GALES/GALES/bin/output/

**Comment by jorvis** [comment]
@JenHauen16 , can you please create a new ticket for this issue.  It seems to be a different one, and the full-path issues from this ticket were resolved in commit 1165f3a42f9f0a23f2b41e00de3d8843d2396fdf

Closing this one.

**Comment by JenHauen16** [comment]
Which files are supposed to be in there?
